apps/serdes/Deserializer.cpp

```diff
@@ -0,0 +1,608 @@
+        auto name = deserialize_string(allocate_stmt->name());
+        auto type = deserialize_type(allocate_stmt->type());
+        Halide::MemoryType memory_type = Halide::MemoryType::Auto;
+        switch (allocate_stmt->memory_type()) {
+        case Halide::Serdes::MemoryType::MemoryType_Auto:
+            memory_type = Halide::MemoryType::Auto;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_Heap:
+            memory_type = Halide::MemoryType::Heap;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_Stack:
+            memory_type = Halide::MemoryType::Stack;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_Register:
+            memory_type = Halide::MemoryType::Register;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_GPUShared:
+            memory_type = Halide::MemoryType::GPUShared;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_GPUTexture:
+            memory_type = Halide::MemoryType::GPUTexture;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_LockedCache:
+            memory_type = Halide::MemoryType::LockedCache;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_VTCM:
+            memory_type = Halide::MemoryType::VTCM;
+            break;
+        case Halide::Serdes::MemoryType::MemoryType_AMXTile:
+            memory_type = Halide::MemoryType::AMXTile;
+            break;
+        }
+        auto extents_serialized = allocate_stmt->extents();
+        auto extents_types = allocate_stmt->extents_type();
+        std::vector<Expr> extents;
+        for (size_t i = 0; i < extents_serialized->size(); ++i) {
+            auto extent = deserialize_expr(extents_types->Get(i), extents_serialized->Get(i));
+            extents.push_back(extent);
+        }
+        auto condition = deserialize_expr(allocate_stmt->condition_type(), allocate_stmt->condition());
+        auto new_expr = deserialize_expr(allocate_stmt->new_expr_type(), allocate_stmt->new_expr());
+        auto free_function = deserialize_string(allocate_stmt->free_function());
+        auto padding = allocate_stmt->padding();
+        auto body = deserialize_stmt(allocate_stmt->body_type(), allocate_stmt->body());
+        return Halide::Internal::Allocate::make(name, type, memory_type, extents, condition, body, new_expr, free_function, padding);
+    }
+    case Halide::Serdes::Stmt_Free: {
+        const Halide::Serdes::Free *free_stmt = (const Halide::Serdes::Free *)stmt;
+        auto name = deserialize_string(free_stmt->name());
+        return Halide::Internal::Free::make(name);
+    }
+    case Halide::Serdes::Stmt_Realize: {
+        const Halide::Serdes::Realize *realize_stmt = (const Halide::Serdes::Realize *)stmt;
+        auto name = deserialize_string(realize_stmt->name());
+        std::vector<Type> types;
```

Comment:
Here and elsewhere: if we can get the `size()` of the Flatbuffers array we are reading from, we should call `.reserve()` on the `vector` we create, to reduce memory resizing churn, e.g.

```
        std::vector<Type> types;
        types.reserve(realize_stmt->types().size());  // Will this work?
```